dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -823,14 +823,18 @@ HRESULT WINAPI CRecycleBin::InvokeCommand(LPCMINVOKECOMMANDINFO lpcmi)
     HRESULT hr;
     LPSHELLBROWSER lpSB;
     IShellView * lpSV = NULL;
+    WCHAR szDrive[8];
 
     TRACE("%p %p verb %p\n", this, lpcmi, lpcmi->lpVerb);
 
     if (LOWORD(lpcmi->lpVerb) == iIdEmpty)
     {
-        // FIXME
-        // path & flags
-        hr = SHEmptyRecycleBinW(lpcmi->hwnd, L"C:\\", 0);
+        if (GetEnvironmentVariableW(L"SystemDrive", szDrive, _countof(szDrive) - 1))
+            PathAddBackslashW(szDrive);
+        else
+            StringCbCopyW(szDrive, sizeof(szDrive), L"C:\\");
```

Comment:
Shouldn't we just fail if there is no such environment variable?
This fallback looks like a hack

Reply:
Also there is the possibility that the recycle bin is on a mounted volume that doesn't have any drive letter, and theoretically they should still be displayed in the recycle bin thing.

Reply:
???